cmd/entrypoint/envoy.go

```diff
@@ -83,16 +84,33 @@ func runEnvoy(ctx context.Context, envoyHUP chan os.Signal) error {
 				defer cancel()
 				wait := subcommand(tctx, "docker", append([]string{"wait"}, cids...)...)
 				wait.Stdout = nil
-				logExecError(ctx, "docker wait", wait.Run())
+				if err := wait.Run(); err != nil {
+					var exitErr *dexec.ExitError
+					if errors.As(err, &exitErr) {
+						dlog.Errorf(ctx, "docker wait: %v\n%s", err, string(exitErr.Stderr))
+					} else {
+						return err
+					}
+				}
```

Comment:
It'd be a weird function since one branch returns but the other continues.

```go
func handleExecErr(err error) error {
	var exitErr *dexec.ExitError
	if !errors.As(err, &exitErr) {
		return err
	}
	dlog.Errorf(ctx, "docker wait: %v\n%s", err, string(exitErr.Stderr))
	return nil
}
```

and then

```go
if err := handleExecErr(wait.Run()); err != nil {
	return err
}
```

and IMO that just makes the whole thing less clear, since it removes from the caller visibility in to when we consider things fatal or non-fatal.